### ISSUE TITLE
Rack Layers behaviour 

### ISSUE THREAD

**Issue by magdesign** [magdesign/IQSynth#3]
Currently medias in the Rack are always Multiplied with each other.
Its not possible to play a media containing black when a Layer below is also running.

Idea:
Standard media plays as is, top Layers always stay on top, if you want to Multiply it,
1.) add a LumaKey FX
or
2.) buttons on each row next to the Opacity slider to select: Multiply, Addition, Subtraction.

![layers_stack](https://user-images.githubusercontent.com/5273166/79876371-3ea86d00-83eb-11ea-9674-9a0b6eaea6ca.png)

**Comment by johnday-github** [comment]
There used to be a few built in options for mixer layers, but I moved this into 'Mixer' type plugins instead. (currently plugins - Layers1 & Layers2)

1. because I wanted to keep/limit the default behavior for simplicity (screen the layers like projectors would)
2. I kept coming up with different options I might want, in this case we can now just make a new mixer plugin for the effect we want
3. and its now possible to store a 'default' startup setup if you always want a particular fixed setup

**Comment by magdesign** [comment]
Still a bit struggling with the Layers Plugin....

How could I make a default startup file where the Layers are handled as Layers (top Layer will complete hide all Layers below if Opacity is 100%) ?

**Comment by johnday-github** [comment]
All a little unused

The layers plugin (as with most of them) could do with some docs. Actually thought I had somewhere described this, but doesn't look like.

Mixer Plugins (or colour coded blue in the plugins area) they have 2 limitations 
1. They will only work in the first slot (F1) in an FX Stack
2. Only work for Buses and Output slots.

To achieve what you want:

Add the "Layers 2" plugin into 'F1' of the bus
- set all the layers to 'Replace Mode' (this will overwrite the layer below, taking into account transparency)
- when in 'Replace Mode' the parameter slider becomes a simple Luma threshold option

To store this a default setup.

The 'default' setup option is really new, I was going to use it the first time at my next gig.

It's currently hidden in:

Configuration & Tools
- Configuration - User - Store current configuration

Then if enabled should on next startup preload all slots and fx that where loaded at the time the button was pressed.

**Comment by magdesign** [comment]
Just tested with 2 videos, both have the EYE Layers 2 on F1 set to Replace.
But the output still shows both (IQSynth 0.16.12).

See this pic, the circle thing should not be visible now...

![layers_fx](https://user-images.githubusercontent.com/5273166/80153123-428be900-85bd-11ea-8b1a-c9f46c645882.png)

**Comment by johnday-github** [comment]
you need to put the plugin in the Bus or Output not a layer

**Comment by johnday-github** [comment]
I'll make it so it won't allow you to drop a mixer plugin into the wrong place, add a warning popup. in next release.

**Comment by johnday-github** [comment]
I've done that now, will be in next release. Actually you can multiple mixer plugins in a Bus or Output.

**Comment by johnday-github** [comment]
Change released in 0.16.13

**Comment by magdesign** [comment]
"Bus or Output" where do I find these ?

**Comment by johnday-github** [comment]
The Output|Screens is the primary display, in the top/center of the display: 
https://iqsynth.com/manual/screens

The Buses are to the left & right sides at the top of the display:
https://iqsynth.com/manual/buses

**Comment by magdesign** [comment]
Good to know where the BUSES are (marked with an orange rectangel) and that it is possible to add FX on them to.

Dropping the "EYE Layers 2" on F1 of the BUS and setting the knobs to 2/5 Replaced did exactly what I need:

![buses](https://user-images.githubusercontent.com/5273166/80317682-7a856d00-8805-11ea-9551-12f40922ad60.png)

**Comment by magdesign** [comment]
Be aware that its only possible to switch modes when there are two medias in the slot.
If there are no medias you are not able to switch to "2/5 Replaced"

**Comment by johnday-github** [comment]
I've just visited this, it might be more of the fact that if you don't have any layers playing you don't see any changes made to the plugin

**Comment by magdesign** [comment]
The Layer behavior is still a bit weird, shouldn't it fade when I adjust the Layer opacity ?
 
![layer](https://user-images.githubusercontent.com/5273166/80319123-f801ab00-880e-11ea-977a-dbd106e69298.gif)

**Comment by johnday-github** [comment]
your looking at a slot, not really a layer at that point, if a blacked out all the slots previews in the layer when you fade the layer, then you would have no idea what you are about to bring in

**Comment by johnday-github** [comment]
the layer controls on the edge should be considered more part of a bus then a slot, maybe I could indicate this by visually detaching the layers from the rack (the rack is not always in a layer mode)

**Comment by magdesign** [comment]
so there is no solution so far to adjust the opacity in ?
Still a bit confused with layers, buses and slots....

The idea is to have 2 videos, the one on the toplayer will fade in/out to the sound or bpm or midi while the lower layer will only be visible when the upper one fades out:

![layer_thing](https://user-images.githubusercontent.com/5273166/80398915-aca3d700-88b8-11ea-95ac-6ee89d7847b2.gif)

**Comment by johnday-github** [comment]
yes, your right that isn't possible at the moment. There is a reason that I fade to black rather then transparent in the default setup (it was quite a deep reason at some time). I think the easiest thing for me to do would be to create a new mixer plugin which has this transparent layer fading option.

**Comment by johnday-github** [comment]
IIRC it was the visual look/feel of the interaction between screen mixing (like overlaying projectors) and fading using transparency and allowing multiple types of mixer setup was why I settled on the fade to black.  I'll get a new mixer out for you shortly.

**Comment by johnday-github** [comment]
Well, I revisited the default layer mixing and it really does have to behave like that.
I'd left those mini layer controls out of the interface for a LONG time, due to this possible confusion when using a different mixer, but in the end I thought it better to have a basic options already in place as I was forever looking for a quick fade out. 

However I've made a nice new Mixer plugin 'MatteLayers' which I think should do what you want. It was a little more complex then I was expecting, making sure it respected transparency, but I'm pretty happy with the result. Pick it up in the new Plugin Collection.
